python-package/lightgbm/sklearn.py

```diff
@@ -673,6 +673,18 @@ def _more_tags(self) -> Dict[str, Any]:
             },
         }
 
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        more_tags = self._more_tags()
+        tags.input_tags.allow_nan = more_tags.pop("allow_nan", False)
+        tagged_input_types = more_tags.pop("X_types", [])
+        tags.input_tags.sparse = "sparse" in tagged_input_types
+        tags.target_tags.one_d_labels = "1dlabels" in tagged_input_types
+        tags._xfail_checks = more_tags.pop("_xfail_checks", {})
+        if more_tags or set(tagged_input_types).difference({"2darray", "sparse", "1dlabels"}):
+            _log_warning(f"Some tags sklearn tag values are missing from __sklearn_tags__: `{more_tags}`")
```

Comment:
I think a warning seen by the user coming from here would be quite confusing. Probably, having a test doing the same check as here is more suitable as you'd be sure it doesn't happen when the user uses the class.

Reply:
Yeah good point, this one is not for the end users but for the maintainers who would touch _more_tags but not update __sklearn_tags__ accordingly!

Reply:
I agree with @adrinjalali , please remove this warning.

Reply:
Removed =)